cla_backend/libs/eligibility_calculator/calculator.py

```diff
@@ -176,23 +177,22 @@ def is_gross_income_complete(case_data):
 
     @staticmethod
     def _translate_section_disposable_income(case_data, request_data):
+        def is_deductions_complete(deductions):
+            for key in deductions.PROPERTY_META:
+                if getattr(deductions, key) is None:
+                    return False
+            return True
+
         """
         Determine if the questions for disposable income section of the test have been completed by the user yet,
         and put this in the CFE request.
         """
 
         def is_disposable_income_complete(case_data):
-            if not hasattr(case_data, "you"):
+            if not is_deductions_complete(case_data.you.deductions):
                 return False
-            person = case_data.you
-            if not hasattr(person, "deductions"):
-                return False
-            deductions = case_data.you.deductions
-            for key in deductions.PROPERTY_META:
-                if not hasattr(deductions, key):
-                    return False
 
-            if not hasattr(case_data.facts, "has_partner"):
+            if case_data.facts.has_partner and not is_deductions_complete(case_data.partner.deductions):
```

Comment:
We've lost the logic that if `has_partner == None` then we need to return False, for the reason in the comment.

`if case_data.facts.has_partner and not is_deductions_complete(case_data.partner.deductions)` is a welcome improvement, but suggest it is added in addition to the existing logic.